src/library/dlgtrackinfo.cpp

```diff
@@ -445,19 +468,58 @@ void DlgTrackInfo::clear() {
 }
 
 void DlgTrackInfo::slotBpmDouble() {
-    spinBpm->setValue(spinBpm->value() * 2.0);
+    m_pBeatsClone->scale(Beats::DOUBLE);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
 }
 
 void DlgTrackInfo::slotBpmHalve() {
-    spinBpm->setValue(spinBpm->value() / 2.0);
+    m_pBeatsClone->scale(Beats::HALVE);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
 }
 
 void DlgTrackInfo::slotBpmTwoThirds() {
-    spinBpm->setValue(spinBpm->value() * (2./3.));
+    m_pBeatsClone->scale(Beats::Beats::TWOTHIRDS);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
 }
 
 void DlgTrackInfo::slotBpmThreeFourth() {
-    spinBpm->setValue(spinBpm->value() * (3./4.));
+    m_pBeatsClone->scale(Beats::Beats::THREEFOURTHS);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
+}
+
+void DlgTrackInfo::slotBpmClear() {
+    spinBpm->setValue(0);
+    m_pBeatsClone.clear();
+
+    bpmConst->setChecked(true);
+    bpmConst->setEnabled(m_trackHasBeatMap);
+    spinBpm->setEnabled(true);
+    bpmTap->setEnabled(true);
+}
+
+void DlgTrackInfo::slotBpmConstChanged(int state) {
+    if (state != Qt::Unchecked) {
+        // const beatgrid requested
+        if (spinBpm->value() > 0) {
+            m_pBeatsClone = BeatFactory::makeBeatGrid(m_pLoadedTrack.data(),
```

Comment:
Do you mean the beat detector unreliably detects the first beat?

If the user has used the "translate" button to align their beat grid then typically they would do that on the first beat, right? That's what I do at least. It seems to me this potentially throws away information if the user put effort into tweaking their beat grid.